geomfn: match PostGIS slightly more with ST_Intersection

otan opened the pull request:
Refs #54429. 

ST_Intersection returns the empty element of one side instead of
whatever GEOS returns if any side is empty.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/54435)
<!-- Reviewable:end -->

otan committed:
geomfn: match PostGIS slightly more with ST_Intersection

ST_Intersection returns the empty element of one side instead of
whatever GEOS returns if any side is empty.

Release note: None

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/54435#-:-MHR7JwK5baix_Y8jgmg:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
thanks!

bors r=sumeerbhola
